### PR TITLE
production updates, improve error handling, configurability

### PR DESCRIPTION
**Pull request by thepiwo**

### CODE REVIEW

File: src/logic.js [hunk 1]
```diff
@@ -58,7 +62,7 @@ function filterTx(tx) {
 
 async function indexContract(ownerId, height) {
   const contractAddress = await client.getAccount(ownerId).then(({ contractId }) => contractId);
-  const contractInstance = await client.getContractInstance({ source: CONTRACT_SOURCE, contractAddress });
+  const contractInstance = await client.getContractInstance({ aci: CONTRACT_ACI, contractAddress });
```
Comment (marc0olo) [inline src/logic.js:65]: yes 😎

File: src/logic.js [hunk 2]
```diff
@@ -33,11 +38,10 @@ const initWebsocket = () => {
 const initClient = async () => {
   if (!client) {
     client = new AeSdk({
-      compilerUrl: 'https://compiler.aepps.com',
```
Comment (marc0olo) [inline src/logic.js:36]: 😎

**Review by marc0olo — APPROVED** [review]
mainly checked the contract relevant stuff. will let @kenodressel judge the rest, hehe

lgtm and good to see that we also get rid of http compiler here 👍

**Comment by marc0olo** [comment]
@thepiwo this will break support with GA contract version `1.0.0`, right? but probably doesn't matter, just to clarify

**Comment by thepiwo** [comment]
@marc0olo no it works for both v1 and v2, there has been no change to the specific entrypoints used :)

**Review by kenodressel — APPROVED** [review]

**Pull request merged**